packages/x-date-pickers/src/DateField/DateField.tsx

```diff
@@ -0,0 +1,22 @@
+import * as React from 'react';
+import TextField from '@mui/material/TextField';
+
+import { DateFieldProps } from './DateField.interfaces';
+import { useDateField } from './useDateField';
+
+type DateFieldComponent = (<TInputDate, TDate = TInputDate>(
+  props: DateFieldProps<TInputDate, TDate> & React.RefAttributes<HTMLInputElement>,
+) => JSX.Element) & { propTypes?: any };
+
+export const DateField = React.forwardRef(function DateField<TInputDate, TDate = TInputDate>(
+  inProps: DateFieldProps<TInputDate, TDate>,
+  ref: React.Ref<HTMLInputElement>,
+) {
+  const { inputRef, inputProps } = useDateField<
+    TInputDate,
+    TDate,
+    DateFieldProps<TInputDate, TDate>
+  >(inProps);
+
+  return <TextField ref={ref} inputRef={inputRef} {...inputProps} />;
+}) as DateFieldComponent;
```

Comment:
Looking at the code, I almost feel like it would be better to expose this userland. Not sure.

Reply:
It is exposed and should be documented soon

Do you think we should not expose the component ?

Reply:
OK great to know.

> Do you think we should not expose the component ?

I can't make my mind on this. I think that it's not impossible that we could be better off without this abstraction, with the rationale that lower level API makes easier customizations.

As a side note, I wonder if developers won't expect to be able to target `MuiDateField` in the theme.

Reply:
https://github.com/mui/mui-x/blob/next/packages/x-date-pickers/src/DateField/DateField.tsx

The component has a little more logic know and supports theme overrides as you mentionned.
And we have multi-input range fields with a lot more logic (those I would clearly not remove them).